src/flint/types/fmpq_mpoly.pyx

```
@@ -0,0 +1,600 @@
+        else:
+            return bool(self - other)
+
+    def context(self):
+        return self.ctx
+
+    def __len__(self):
+        return fmpq_mpoly_length(self.val, self.ctx.val)
+
+    def coefficient(self, slong i):
+        cdef fmpq v
+        if i < 0 or i >= fmpq_mpoly_length(self.val, self.ctx.val):
+            return fmpq(0)
+        else:
+            v = fmpq.__new__(fmpz)
+            fmpq_mpoly_get_term_coeff_fmpq(v.val, self.val, i, self.ctx.val)
+            return v
+
+    def exponent_tuple(self, slong i):
+        cdef slong j, nvars
+        cdef fmpz_struct ** tmp
+        if i < 0 or i >= fmpq_mpoly_length(self.val, self.ctx.val):
+            raise ValueError
+        nvars = self.ctx.nvars()
+        res = tuple(fmpz() for j in range(nvars))
+        tmp = <fmpz_struct **> libc.stdlib.malloc(nvars * sizeof(fmpz_struct **))
+        try:
+            for j in range(nvars):
+                tmp[j] = &((<fmpz> (res[j])).val[0])
+            fmpq_mpoly_get_term_exp_fmpz(tmp, self.val, i, self.ctx.val)
+        finally:
+            libc.stdlib.free(tmp)
+        return res
+
+    def repr(self):
+        return self.str() + "  (nvars=%s, ordering=%s names=%s)" % (self.ctx.nvars(), self.ctx.ordering(), self.ctx.py_names)
+
+    def str(self):
+        cdef char * s = fmpq_mpoly_get_str_pretty(self.val, self.ctx.c_names, self.ctx.val)
+        try:
+            res = str_from_chars(s)
+        finally:
+            libc.stdlib.free(s)
+        if res.startswith(" - "):
+            res = "-" + res[3:]
+        return res
+
+    def __neg__(self):
+        cdef fmpq_mpoly res
+        res = create_fmpq_mpoly(self.ctx)
+        fmpq_mpoly_neg(res.val, (<fmpq_mpoly>self).val, res.ctx.val)
+        return res
+
+    def __add__(self, other):
+        cdef fmpq_mpoly res
+        if typecheck(other, fmpq_mpoly):
+            if (<fmpq_mpoly>self).ctx is not (<fmpq_mpoly>other).ctx:
+                return NotImplemented
```

As discussed this should raise an error.